jiant/utils/options.py

```diff
@@ -19,22 +20,25 @@ def parse_task_list_arg(task_list):
     return task_names
 
 
-def parse_cuda_list_arg(cuda_list):
+def parse_cuda_list_arg(cuda_arg):
     """
     Parse cuda_list settings
     """
     result_cuda = []
-    if cuda_list == "auto":
-        cuda_list = list(range(torch.cuda.device_count()))
-        return cuda_list
-    elif isinstance(cuda_list, int):
-        return cuda_list
-    elif "," in cuda_list:
-        return [int(d) for d in cuda_list.split(",")]
+    if cuda_arg == "auto":
+        result_cuda = list(range(torch.cuda.device_count()))
+        if len(result_cuda) == 1:
+            result_cuda = result_cuda[0]
+    elif isinstance(cuda_arg, int):
+        result_cuda = cuda_arg
+    elif "," in cuda_arg:
+        result_cuda = [int(d) for d in cuda_arg.split(",")]
     else:
         raise ValueError(
             "Your cuda settings do not match any of the possibilities in defaults.conf"
         )
-    if len(result_cuda) == 1:
-        result_cuda = result_cuda[0]
+    if torch.cuda.device_count() == 0:
+        # If there are no cuda devices available, default to -1.
+        log.info("CUDA devices not found, defaulting to cuda=-1")
+        result_cuda = -1
```

Comment:
how do we feel about failing / louder warning if user thinks they should be using CUDA but devices aren't found ?

Reply:
we can do that.